### ISSUE TITLE
logMaxSize global param seems to be defunct

### ISSUE THREAD

**Issue by rgerhards** [buildbot/buildbot#4504]
The `logMaxSize` global parameter (and friends) seem to be defunctional in buildbot current and at least also in 1.1.

We have issues with some CI tests that generate excessively large log files in some error cases. This leads to Out Of Memory on the buildbot master machine and in turn often leads to total machine hangup. Configuration is
```
c = BuildmasterConfig = {}
...
# Global Log Settings
c['logCompressionMethod'] = 'gz'
c['logMaxSize'] = 1024*1024*32 # 32MiB
c['logMaxTailSize'] = 64*1024
c['logEncoding'] = 'utf-8'
```
Nevertheless, buildbot accepts far longer logs, Here is a sample of a ~580MiB log generated while these settings were active: https://build.rsyslog.com/#/builders/181/builds/909

Grepping over buildbot current (master branch in this case) shows that `logMaxSize` is read from config but never used. Commit https://github.com/buildbot/buildbot/commit/b8ccc08d4bb4e80244503cf37b6c8e73fb764f48 introduced the global parameter. However, buildbot current seems to not have any remains of that code. Might it have been removed accidently in the mean time. I tried to track this down in git history, but not being a Python programmer I could not really find a trace of where the code was moved to (or lost).

I would appreciate if someone could have a look at this issue.

**Comment by tardyp** [comment]
Indeed, this is fairly possible that tis config option has been overlooked during conversion to data api.
would you like to contribute a fix to that?

I think the most straighforward way to implement this is to count logged bytes here:
https://github.com/buildbot/buildbot/blob/3ea4fdb113c58093c662562bf75c1be385e5d33e/master/buildbot/process/log.py#L79

You would just count the number of chars passed here, and stop calling self.master.data.updates.appendLog when the maxLogSize has been reached.

**Comment by rgerhards** [comment]
@tardyp I have crafted a patch in https://github.com/buildbot/buildbot/pull/4505 - but it looks like it is not 100% correct. On my system, when I have large logfiles, I get many calls to addRawLine with buffer sizes slightly above 70000 bytes. I hit truncated state, then I continue to get maybe ~100..200 calls to addRawLines and then the buildbot process goes away. No failure indication anywhere, neither in the system log nor in twistd.log. From monitoring, I know for sure that memory is keeping low, so excessive mem use cannot be the culprit. Did I overlook something with the patch? Or do you have a suggestion how I could try to find what makes the process go away. I admit I am totally stumbled right now.

**Comment by jpflori** [comment]
Is there any help needed on that one?

**Comment by tardyp** [comment]
Any help is appreciated...
We can restore this feature of this is needed for you.
Buildbot could drop the lines of a threshold is crossed

Le mar. 6 avr. 2021 à 20:13, Jean-Pierre Flori ***@***.***> a
écrit :
